xet_threadpool/src/threadpool.rs

```diff
@@ -151,55 +150,21 @@ impl ThreadPool {
     /// from within a task running on the runtime worker pool.  Doing so can lead to deadlocking.
     pub fn external_run_async_task<F>(&self, future: F) -> Result<F::Output, MultithreadedRuntimeError>
     where
-        F: std::future::Future + Send + 'static,
+        F: Future + Send + 'static,
         F::Output: Send + Sync,
     {
         self.external_executor_count.fetch_add(1, Ordering::SeqCst);
 
         let ret = self.handle.block_on(async move {
             // Run the actual task on a task worker thread so we can get back information
             // on issues, including reporting panics as runtime errors.
-            self.handle.spawn(future).await.map_err(|e| {
-                if e.is_panic() {
-                    // The task panic'd.  Pass this exception on.
-                    error!("Panic reported on xet worker task: {e:?}");
-                    MultithreadedRuntimeError::TaskPanic(e)
-                } else if e.is_cancelled() {
-                    // Likely caused by the runtime shutting down (e.g. with a keyboard CTRL-C).
-                    MultithreadedRuntimeError::TaskCanceled(format!("{e}"))
-                } else {
-                    MultithreadedRuntimeError::Other(format!("task join error: {e}"))
-                }
-            })
+            self.handle.spawn(future).await.map_err(MultithreadedRuntimeError::from)
```

Comment:
The default from method doesn't distinguish between cancelled and panic, so without the above code CTRL-C will show up as a panic runtime error in python instead of a TaskCancellation exception, which is handled differently.  If this code isn't part of the key path, let's not change it.